Add unified exec escalation handling and tests

pakrym-oai opened the pull request:
Similar implementation to the shell tool

pakrym-oai committed:
Support unified exec escalation approvals

pakrym-oai committed:
Update dependencies in Cargo.lock: bump quick-xml to 0.38.0, bitflags to 2.10.0, and thiserror to 2.0.17

core/tests/suite/approvals.rs: workaround Linux sandbox arg0 test on ARM by using ReadOnly policy and cfg-gating test scenario

pakrym-oai committed:
approvals: adjust unified exec scenario for Linux aarch64 sandbox and use python3 test command

dylan-hurd-oai commented on codex-rs/core/src/tools/handlers/unified_exec.rs:
```diff
@@ -100,8 +104,30 @@ impl ToolHandler for UnifiedExecHandler {
+                    )
+                {
+                    return Err(FunctionCallError::RespondToModel(format!(
+                        "approval policy is {policy:?}; reject command — you should not ask for escalated permissions if the approval policy is {policy:?}",
```
```suggestion
                        "approval policy is {policy:?}; reject command — you cannot ask for escalated permissions if the approval policy is {policy:?}",
```

dylan-hurd-oai replied:
non-blocking suggestion

dylan-hurd-oai reviewed (commented):
Going to defer to @jif-oai for better context on the new tool framework

pakrym-oai committed:
codex-rs: clarify error message wording and update justification type

- Update error message to say "you cannot ask for escalated permissions" instead of "you should not ask"
- Change justification in ExecCommandRequest from Option<&str> to Option<String>
- Propagate String justification throughout unified_exec and session_manager
